### ISSUE TITLE
Installation via Swift Packages Manager

### ISSUE THREAD

**Issue by gsgod** [TextureGroup/Texture#1714]
Hi. 
Is there a support for swift packages manager in texture ?
it would be great if we could support the Swift Packages Manager.

**Issue Description**
I'm trying to install master branch of Texture via Swift Packages Manager but fails. 
Seems a problem with manifest.

<img width="716" alt="스크린샷 2019-10-25 오전 9 39 10" src="https://user-images.githubusercontent.com/3404013/67536207-f09eff00-f710-11e9-817b-1246b89d1476.png">

**Comment by SergeyPetrachkov** [comment]
Well, is not SPM mostly for swift? And Texture is all about Objc. So...

**Comment by xezero** [comment]
Any updates on this? Swift Package Manager is the future! Would love to see the integration via SPM supported.

**Comment by pcabfever** [comment]
Yeah, I totally agree with @xezero, is SPM support in the roadmap of this fantastic framework?

**Comment by PabloLerma** [comment]
I'd be very helpful for us having this framework in SPM 🙏

**Comment by Anhelme** [comment]
Any updates?

**Comment by xezero** [comment]
Now that we have Xcode 12 GM with more SPM improvements, any efforts or updates on this potentially?

**Comment by SergeyPetrachkov** [comment]
Would be interesting and challenging at the same time. SPM only supports pure packages which means it can support ObjC or C/C++ code but when it's not mixed up inside the package. But SPM also supports binaries now. So maybe the goal can be achieved by distributing binaries through SPM.

**Comment by schazers** [comment]
+1 for this!

**Comment by CodeEagle** [comment]
Two steps to add support for SPM for Texture:
1. Generate `AsyncDisplayKit.xcframework.zip` when release a new version and drag it to 
<img width="1161" alt="image" src="https://user-images.githubusercontent.com/2311352/106695705-1b674b00-6616-11eb-9fe8-260e0c711e0e.png">
2. add `Package.swift` point to the zip file like [`this`](https://github.com/CodeEagle/Texture/blob/main/Package.swift)

**Comment by that-scalcucci-guy** [comment]
+1 as well for official support. I'm aware of the frustrating limitations of SPM (as I have to wrap quite a bit of my C++ code as well).

As @CodeEagle  and @SergeyPetrachkov have said, it has recently become much easier to accomplish.

**Comment by larsschwegmann** [comment]
+1

**Comment by maximkrouk** [comment]
+1

**Comment by deep-randhawa** [comment]
+1

**Comment by Mr-Sergey-K** [comment]
+1

**Comment by rogerluan** [comment]
Any word on this? Perhaps @rcancro ? 🙏 😊

**Comment by elilien** [comment]
+1

**Comment by johncodeos** [comment]
+1

**Comment by babbage** [comment]
> Would be interesting and challenging at the same time. SPM only supports pure packages which means it can support ObjC or C/C++ code but when it's not mixed up inside the package. But SPM also supports binaries now. So maybe the goal can be achieved by distributing binaries through SPM.

I was surprised to learn recently, through necessity, that it _is_ possible to mix different languages, when the code can be packaged into separately targets that are each purely Swift, or purely Objective-C, and so on. These can then be added as dependencies of an overall Package that is provided to end users. This was news to me, proved easy to set up, and works well. Just in case that helps get this made available via SPM.

**Comment by rcancro** [comment]
This is not a priority for us. It would be a great contribution from the community though!

**Comment by shimastripe** [comment]
As a workaround, I have an SPM package based on 3.1.0 with XCFramework binaries. Please refer to this.
https://github.com/shimastripe/Texture/releases/tag/3.1.1

**Comment by ivanmoskalev** [comment]
I'm actually willing to do this. Will get to it in a week. Please let me know via reactions if there is yet any demand :)

**Comment by SergeyPetrachkov** [comment]
> I'm actually willing to do this. Will get to it in a week. Please let me know via reactions if there is yet any demand :)

Are you gonna try to automate the process and add an official SPM integration into this repo or do a fork? :) 

On a side note, I wonder how alive this project is and how well it plays along with the UIKit that now has MainActor isolation by default and all other new shiny toys that iOS and macOS SDKs have

**Comment by vBoykoGit** [comment]
+1

**Comment by trungnguyen1791** [comment]
+1

Do we have any goodnews with SPM integration guys? Especially when Cocoapods soon to be read-only...

**Comment by gsgod** [comment]
[TextureSwiftSupport](https://github.com/FluidGroup/TextureSwiftSupport) 

It looks like this extension library should address the issue, so I’ll close this issue. Thank you to everyone who has followed and contributed to the discussion over the years. Happy coding!
